codepropertygraph/src/main/scala/io/shiftleft/passes/CpgPass.scala

```diff
@@ -121,6 +121,7 @@ object CpgPass {
   * diff graph, which additionally provides a map from nodes to graph ids.
   * */
 case class AppliedDiffGraph(diffGraph: DiffGraph,
+                            inverseDiffGraph: Option[DiffGraph],
```

Comment:
We are now storing a `diffGraph` along with the `inverseDiffGraph`. Is this necessary or would it be possible to generate the `inverseDiffGraph` on the fly in the undo operation? I can imagine you store the inverseDiffGraph here so that you can see what was in the properties before the operation, is that why it's done in this way?

Reply:
I don't need original diffgraph, this is just for returning multiple values from the function.
It is possible to generate undo from original DiffGraph but then it would need to store old property values and edge references in vars which are set when it is applied which would make it a bit more complicated.

Reply:
So, this means that when we call `applyDiff` with `undoable = true`, we end up with the original diff graph in `diffGraph` and its inverse in `inverseDiffGraph`, correct? That means, we hold the diffgraph along with its inverse in memory at once?

Reply:
Yes, but you should discard original diffgraph if it is no longer needed and just take the appliedDiff.inverseDiffGraph to keep it in memory until you will need the undo functionality.

Reply:
Ah, I see, yes, that makes sense. And we'll always have the situation where we briefly hold both, that's something that we won't get around. Ok, approved, from my side.